### ISSUE TITLE
UserWarning: Just-in-time loading and compiling the CUDA kernels of SRU was unsuccessful.

### ISSUE THREAD

**Issue by wcycqjy** [spkgyk/TDFNet#2]
Hi, when I run the training codes from RTFS-Net, the error shows:
```
UserWarning: Just-in-time loading and compiling the CUDA kernels of SRU was unsuccessful. Got the following error:
Error building extension 'sru_cuda': [1/3] C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\bin\nvcc --generate-dependencies-with-compile --dependency-output sru_cuda_kernel.cuda.o.d -Xcudafe --diag_suppress=dll_interface_conflict_dllexport_assumed -Xcudafe --diag_suppress=dll_interface_conflict_none_assumed -Xcudafe --diag_suppress=field_without_dll_interface -Xcudafe --diag_suppress=base_class_has_different_dll_interface -Xcompiler /EHsc -Xcompiler /wd4190 -Xcompiler /wd4018 -Xcompiler /wd4275 -Xcompiler /wd4267 -Xcompiler /wd4244 -Xcompiler /wd4251 -Xcompiler /wd4819 -Xcompiler /MD -DTORCH_EXTENSION_NAME=sru_cuda -DTORCH_API_INCLUDE_EXTENSION_H -ID:\ProgramData\miniforge3\envs\TDFNet\Lib\site-packages\torch\include -ID:\ProgramData\miniforge3\envs\TDFNet\Lib\site-packages\torch\include\torch\csrc\api\include -ID:\ProgramData\miniforge3\envs\TDFNet\Lib\site-packages\torch\include\TH -ID:\ProgramData\miniforge3\envs\TDFNet\Lib\site-packages\torch\include\THC "-IC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\include" -ID:\ProgramData\miniforge3\envs\TDFNet\Include -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_86,code=compute_86 -gencode=arch=compute_86,code=sm_86 -c D:\ProgramData\miniforge3\envs\TDFNet\Lib\site-packages\sru\csrc\sru_cuda_kernel.cu -o sru_cuda_kernel.cuda.o 
FAILED: sru_cuda_kernel.cuda.o 
```

Do you know why this happens? I searched on Internet and someone says it's because cuda version lower than 10.2 doesn't have --generate-dependencies-with-compile argument, but my cuda version is 11.0.

**Comment by spkgyk** [comment]
This usually happens because you haven't correctly setup the environment variables in your system. Try following the instructions here 

https://pypi.org/project/sru/#description

**Comment by wcycqjy** [comment]
I indeed installed it by following instructions on this website, such as installing it from the source. But this error still occurs.

**Comment by wcycqjy** [comment]
I find the error occurs due to that ninja/nvcc tries to build the code by using `--sm_86` which is supported by cuda after version `11.2`, but my cuda is `11.0` This was fixed after I changed my cuda toolkit from `11.0` to `11.7`. Similar to [https://github.com/asappresearch/sru/issues/195]
